### ISSUE TITLE
Trim does not work as expected for binary! argument

### ISSUE THREAD

**Issue by loziniak** [red/red#3873]
**Describe the bug**
Asuming that `trim` should erase zero bytes from `binary!` values, it's acting differently than with `string!`'s spaces.

Actual:
```
>> trim #{000012340000}
== #{12340000}
>> trim/tail #{000012340000}
== #{}
```
should work similarly as with `string!`:
```
>> trim "   123   "
== "123"
>> trim/tail "   123   "
== "   123"
```

**Platform version (please complete the following information)**
```
Red 0.6.4 for Linux built 25-Mar-2019/12:11:08+02:00 commit #e6c96aa
```

**Comment by endo64** [comment]
Might not be directly related but better fix together with #3663

**Comment by greggirwin** [comment]
We clearly needs some tests for `trim` on `binary!` values. I think there are a couple errors in the logic, e.g., when `/tail` is used, but `/head` isn't. It should check against `tail` here in `trim-head-tail`:
```
while [
				all [head < tail head/value = null-byte]
			]
```
But if neither refinement is used, it never marks the tail. That is, string has this logic block, which binary lacks entirely:
```
either all [not head? not tail?] [
...
```
